### ISSUE TITLE
Revamp data fields

### ISSUE THREAD

**Issue by frabyn** [frabyn/nodoc#44]
This is a larger undertaking. Nodoc eventually will have this directory structure.

/trial
/appellate
/collateral-attack
/expunction
/pretrial-writ
/pretrial-writ-appeal

Each of these directories will represent a stage of proceedings in a criminal case. Each stage will define the characteristics of that stage. All people can be reused, and should be stored in a different file that can be referenced.

So I'd like to revamp our data file to represent the following fields, and make corresponding changes to the files.

So now, instead of defining the "case below" in an appeal, we simply can call the cause number of the trial case that is defined. 

The data that will remain the same across all kinds of case is the client information.

Same across all stages:
- Name of client

For the client, the following fields should be defined:
Sex
Race
Date of Birth
Social Security Number
Driver License Number
Address

For the stages:

Trial
- Lead Lawyer
- Secondary Lawyer
- Case Number
- Prosecuting Office
- Prosecutor
- Court
- Offense charged
- Date of offense
- Date of arrest
- County of arrest
- City of arrest
- Arresting agency
- Offense report number
- Case Disposition

Appellate
- Lead Lawyer
- Secondary Lawyer
- Case number
- Prosecuting Office
- Prosecutor
- Court

Collateral Attack
- Case number
- Prosecutor
- Prosecuting Office

Expunction
- Case number
- Court

Pretrial Writ
- Case number
- Court
- Prosecutor
- Prosecuting office

Pretrial Writ Appeal
- Case number
- Court
- Prosecutor
- Prosecuting office

**Comment by AndreyBPopov** [comment]
Hi Franklin!
I don't see the "judge" field above. Is it needed?

**Comment by AndreyBPopov** [comment]
Can you please describe the meaning of "•Prosecuting Office" field?
Is an address or name of Prosecuting Office?

**Comment by AndreyBPopov** [comment]
Is Trial Defense Counsel in texas-coa-brief equals to Lead Lawyer at Trial Stage?

**Comment by AndreyBPopov** [comment]
Also I think that
•Date of arrest
•County of arrest
•City of arrest
•Arresting agency

are more related to person than to trial stage.  What do you think?

**Comment by frabyn** [comment]
In my mind, Date, County, City, and Agency go with the trial case rather than with the person.

A prosecutor's office will be a name and an address. I believe these will _almost always_ be printed together but defining them separately for now may be the better practice. 

So for example: 

Name: Harris County District Attorney
Address: 1201 Franklin Street, 6th Floor
Houston, Texas 77002

I would like to keep prosecutors' office data in the court-data file and then define which office is present in each case.

So for a trial case, we would define "HarrisTXDA" as the trial prosecuting office, name the trial prosecutor and then we could do something like an envelope and letter:

\Print{trial_prosecutor}
\Print{trial_prosecutor_office}
\Print{trial_prosecutor_address}

yields

Peter Persecutor
Harris County District Attorney
1201 Franklin Street, 6th Floor
Houston, Texas 77002

**Comment by frabyn** [comment]
Also, let's define clerk name and clerk address for each stage. Here are csv files of all courts and clerks in the state.

Clerks are defined as "County Clerk Office" in the county.csv file and "District Clerk Office" in the district.xls file. Clerks are listed under the "Court" field, along with courts. Court, County, Address, City, Zip are the fields that we will use.

So for Anderson County, the name is a combination of two fields:

`<county> <court>`
resulting in "Anderson County Clerk Office"

Also these files define all courts that will eventually be used, and at the county level they follow the same format. So for the Anderson County Court at Law:

`<county> <court>`
resulting in "Anderson County Court at Law"

However, the district names are handled differently. For clerk's offices, use the same format. For the Anderson district clerk:

So for the Anderson County Court at Law:

`<county> <court>`
resulting in "Anderson District Clerk Office"

But for district court names, only use the "Court" field for the name. For example, the name of the 96th District Court is "96th District Court" and the location is "Anderson County, Texas"

Courtname: 96th District Court
Court Location: Anderson County, Texas

We will use this on trial pleadings.

For county courts, the location is in the court name:

"Anderson County Court at Law" and on the trial pleading the location would be blank





[courts.zip](https://github.com/frabyn/nodoc/files/703165/courts.zip)

**Comment by AndreyBPopov** [comment]
Hi Franklin!
Thanks for courts.zip! How are you planning to use csv files in them?
Now we keep all courts data in texas-courts.tex. But since you have these csv files there are two ways:
1) I can once convert all data from CSV files to the current format in texas-courts.tex so for each line in CSV command like below will be created.
\SetupCourt{TXcoa3}{
name={Third Court of Appeals of~Texas},
place={Austin},
address={}
}
So from this moment you will not need CSV files and edit all data in tex.

2) The second way is to use some latex packages that can parse CVS files. So all courts data will be actually stored in CSV files. And to use some court in your documents you will need to specify the row number in CSV file, e.g. I will create command like \UseCourt{27} and court that corresponds to 27th row in CVS file will be used.

**Comment by frabyn** [comment]
I prefer the first way. The data will not change often so better to deal with it natively in TeX

**Comment by AndreyBPopov** [comment]
Okay, I will move data from CSV to tex today.

**Comment by AndreyBPopov** [comment]
Hi Franklin!
I have wrote VBA macro that transforms data from your CSV files to TeX format. Please look at
example output attached. However I have some problem: when you created CSV files you used comma ","
as the separator between fields, but address field in CSV file also can contain commas and this leads to incorrect parsing of the CSV file. Even Excel or OpenOffice cannot parse you files correctly, I have checked that. Could you please re-export CSVs with ";" as the separator?

**Comment by AndreyBPopov** [comment]
Seems I can't attach the TeX file here. But I commit it to GitHub. Please look at county.tex in Data folder.

**Comment by frabyn** [comment]
Thank you for your work on county.tex! It works great. I would like to go ahead and make a combined file for all district and county courts, called texas-courts.tex

I have changed my mind just a bit about how to format the data. I had said before that the formatting of the data from county courts should be different than district courts.

Instead, I want them both to be handled the same. For counties, the name of the court in the .tex data file should be the same as it is in the csv file, so for example

```
\SetupCourt{TXcoNNN}{
name={$name},
place={$county County, Texas},
website={$website},
address={$address},
city={$city},
zipcode={$zip}
}
```

The e-mail and phone fields do not need to be put into the tex data file.

**Comment by AndreyBPopov** [comment]
Do you have a possibility re-export CSC with  with ";" as the separator? Old CSV is parsed incorrectly now...

**Comment by frabyn** [comment]
Try these out?
https://bynumlaw.sandcats.io/shared/jME7zUMfMSn72HXas2WaCE8fJU_nhwvf3NuGyOf3Juh

**Comment by AndreyBPopov** [comment]
Hi Franklin!
New CSVs seems are correct. Thank you very much!
I will export all data to tex now.
Since we are trying to make general solution I will split this:
place={$county,  Texas}
to
place={$county},
state={Texas}
also I'm tending to export email and phone too, though you dont need them, just in case.

**Comment by AndreyBPopov** [comment]
I commit the final texas-courts.tex. Please take a look.

**Comment by AndreyBPopov** [comment]
Now it is needed to make relevant changes in style files accordind to your new data fields scheme.

**Comment by frabyn** [comment]
Right, have we done that yet? Also, can you add the county courts and clerks to the texas-courts file?

**Comment by AndreyBPopov** [comment]
All country courts and clerks are already in texas-courts.tex

**Comment by AndreyBPopov** [comment]
it was ab6b5cefe354ef56a301c348b3489ddbfc99e180 commit

**Comment by AndreyBPopov** [comment]
Please note that switching to the new data scheme, i.e. stages, will break correct formatting/displaying in all documents!

**Comment by frabyn** [comment]
No problem, let's go ahead and do it. I see now the county courts, sorry about that.
